.github/workflows/ci.yml

```
@@ -1,80 +1,93 @@
-              run: vendor/bin/phpstan analyze src/ tests/ --level=max
+      - name: Install the dependencies
+        run: composer install --no-interaction
 
-    tests:
-        name: PHP ${{ matrix.php }} / SF ^${{ matrix.symfony }}
-        runs-on: ubuntu-latest
-        strategy:
-            fail-fast: false
-            matrix:
-                php: [7.4, 8.0]
-                symfony: [4.4, 5.4, 6.0]
-                exclude:
-                    # Symfony 6.0 does not supports php <8.0
-                    - php: 7.4
-                      symfony: 6.0
-        steps:
-            - name: Setup PHP
-              uses: shivammathur/setup-php@2.9.0
-              with:
-                  php-version: ${{ matrix.php }}
-                  extensions: dom, fileinfo, filter, gd, hash, intl, json, mbstring, pcre, pdo_mysql, zlib
-                  coverage: none
+      - name: Check the coding style
+        run: vendor/bin/php-cs-fixer fix --diff --dry-run
 
-            - name: Checkout
-              uses: actions/checkout@v2
+      - name: Analyze the code
+        run: vendor/bin/phpstan analyze
 
-            - name: Install the dependencies
-              run: |
-                composer require symfony/framework-bundle:^${{ matrix.symfony }} symfony/http-foundation:^${{ matrix.symfony }}
-                composer install --no-interaction --no-suggest
-            - name: Run the unit tests
-              run: vendor/bin/phpunit --colors=always
+  tests:
+    name: PHP ${{ matrix.php }} / SF ^${{ matrix.symfony }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        php: [8.0, 8.1, 8.2]
+        symfony: [5.4, 6.0, 6.1, 6.2, 6.3, 6.4, 7.0]
+        exclude:
+          - php: 8.0
+            symfony: 6.1
+          - php: 8.0
+            symfony: 6.2
+          - php: 8.0
+            symfony: 6.3
+          - php: 8.0
+            symfony: 6.4
+          - php: 8.0
+            symfony: 7.0
+          - php: 8.1
+            symfony: 7.0
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: dom, fileinfo, filter, gd, hash, intl, json, mbstring, pcre, pdo_mysql, zlib
+          coverage: none
 
-    prefer-lowest:
-        name: Prefer Lowest
-        runs-on: ubuntu-latest
-        steps:
-            - name: Setup PHP
-              uses: shivammathur/setup-php@2.7.0
-              with:
-                  php-version: 7.4
-                  extensions: dom, fileinfo, filter, gd, hash, intl, json, mbstring, pcre, pdo_mysql, zlib
-                  coverage: none
+      - name: Checkout
+        uses: actions/checkout@v3
 
-            - name: Checkout
-              uses: actions/checkout@v2
+      - name: Install the dependencies
+        run: |
+          composer require symfony/framework-bundle:^${{ matrix.symfony }} symfony/http-foundation:^${{ matrix.symfony }}
```

```suggestion
          composer require symfony/framework-bundle:^${{ matrix.symfony }} symfony/http-foundation:^${{ matrix.symfony }} symfony/yaml:^${{ matrix.symfony }}
```

maybe this fixes the unit test 😬

---

It did, nice!